### ISSUE TITLE
_media doesn't evaluate dynamically

### ISSUE THREAD

**Issue by Radiergummi** [lowdefy/lowdefy#799]
The `_media` operator returns information about the current screen dimensions, but doesn't update them if the size changes during client runtime.  
Ideally, the data would change if the window is resized, so block visibility could be controlled by media queries, similar to CSS.

There's mainly two ways this can be achieved in modern browsers:

1. The [`resize` event](https://developer.mozilla.org/en-US/docs/Web/API/Window/resize_event):
   Listening to the event (ideally [debounced](https://github.com/lowdefy/lowdefy/issues/724)), we can set the current screen dimensions after the window has been resized.
2. The [`matchMedia() function`](https://developer.mozilla.org/en-US/docs/Web/API/Window/matchMedia):
   Using `matchMedia`, we can attach CSS media queries to the `window` object and add event listeners that will be triggered if the query starts to match, or doesn't match any longer. This allows to set named breakpoints (eg. `sm`, `lg`, etc.) in the application state.

By implementing those features combined, we could have up-to-date media query information available.

**Comment by Gervwyk** [comment]
In order to update block definitions, operators must also be reevaluated in engine. This can be done by calling `context.update();` in the event.  We need to consider where is the best place to register event listeners, I'm sure there would be other event listeners to add in the future besides resize. Perhaps the Events class should include a `addEventListener` method? Not sure, @SamTolmay?

**Comment by SamTolmay** [comment]
I agree, but I don't think that should be part of the `Events` class, since that handles block events where this is more of an engine thing. We also need to be tracking which page is currently being rendered so we don't update the other pages. This could just be a check in `context.update()` though.

**Comment by Gervwyk** [comment]
True also there is an events class for every block, so not what we are looking for. Maybe this should be implemented from `Context` component inside of renderer?

**Comment by Radiergummi** [comment]
Maybe another solution for responsive layouts would be to have additional, size-dependent layout subkeys?
```yaml
layout:
  span: 24
  md:
    span:
      12
  lg:
    span:
      8  
```
The above example would simply add the layout classes with some kind of CSS media query, similar to how TailwindCSS does it. What do you think?

**Comment by SamTolmay** [comment]
This is already a feature that we forgot to document properly 😅. You can see #685 for reference.

**Comment by Radiergummi** [comment]
Haha alright that makes my life easier instantly, thank you 😄

**Comment by Gervwyk** [comment]
Btw, all styles are implemented as [emotion](https://emotion.sh/docs/media-queries) classes. So you should also be able to do:
```
id: bl
type: Box
style:
  '@media (max-width: 420px)':
     display: none
...
```
I have not checked this, but think it'll work. Else try `& > @media (max-width: 420px)`.

**Comment by Gervwyk** [comment]
Sorry, updated / corrected the previous comment.
